### PR TITLE
Add tracing feature to EditSessions

### PR DESCRIPTION
**Pull request by octylFractal**
This helps track down what plugin (or more specifically, extent) is blocking actions from happening.

Suggestions are welcome on every part of this, as it was just the simplest way I could see today. Maybe there's something smarter we can be doing.

I did wrap every extent, just in case we have a bug in one of our own extents. It's better to be safe than sorry.

### CODE REVIEW

File: worldedit-core/src/main/java/com/sk89q/worldedit/EditSessionBuilder.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+/**
+ * A builder-style factory for {@link EditSession EditSessions}.
+ */
+public final class EditSessionBuilder {
+
+    private final EventBus eventBus;
+    @Nullable
+    private World world;
+    private int maxBlocks = -1;
+    private Actor actor;
```
Comment (wizjany) [inline worldedit-core/src/main/java/com/sk89q/worldedit/EditSessionBuilder.java:42]: `@Nullable` actor?

File: worldedit-core/src/main/java/com/sk89q/worldedit/command/WorldEditCommands.java [hunk 2]
```diff
@@ -138,6 +143,45 @@ public void report(Actor actor,
...
+        try (EditSession ignored = session) {
+            if (action.test.test(session, localSession.getPlacementPosition(actor))) {
+                actor.printInfo(TranslatableComponent.builder("worldedit.trace.success")
+                    .args(TextComponent.of(action.toString()))
+                    .build());
+                return;
+            }
+        }
+        List<TracingExtent> tracingExtents = session.getTracingExtents();
+        assert tracingExtents != null;
```
Comment (wizjany) [inline worldedit-core/src/main/java/com/sk89q/worldedit/command/WorldEditCommands.java:166]: leftover debug?

File: worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/AbstractPlayerActor.java [hunk 3]
```diff
@@ -334,7 +334,8 @@ public void floatAt(int x, int y, int z, boolean alwaysGlass) {
             BlockVector3 spot = BlockVector3.at(x, y - 1, z);
             final World world = getWorld();
             if (!world.getBlock(spot).getBlockType().getMaterial().isMovementBlocker()) {
-                try (EditSession session = WorldEdit.getInstance().getEditSessionFactory().getEditSession(world, 1, this)) {
+                try (EditSession session =
```
Comment (wizjany) [inline worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/AbstractPlayerActor.java:337]: is this break necessary? looks like you tried newline/indenting the original line and then converted to the shortcut method.

File: worldedit-core/src/main/java/com/sk89q/worldedit/command/WorldEditCommands.java (hunk 2 above)
Comment (octylFractal) [reply worldedit-core/src/main/java/com/sk89q/worldedit/command/WorldEditCommands.java:166]: No, just convincing IntelliJ that it's not `null`. I can leave it out, I just don't like leaving the warnings in IntelliJ :P

**Pull request merged**